src/irmin/object_graph.ml

```diff
@@ -173,11 +173,11 @@ module Make (Hash : HASH) (Branch : Type.S) = struct
           cache_size);
     let marks = Table.create cache_size in
     let mark key level = Table.add marks key level in
-    let todo = Stack.create () in
+    let todo (* ISSUE *) = Stack.create () in
     (* if a branch is in [min], add the commit it is pointing to too. *)
     Lwt_list.fold_left_s
       (fun acc -> function
-        | `Branch _ as x -> pred x >|= fun c -> (x :: c) @ acc
+        | `Branch _ as x -> pred x >|= fun c -> (* ISSUE *) (x :: c) @ acc
```

Comment:
Should we really keep `x` in min? We're going to stop at it and never visit `pred x`.